### ISSUE TITLE
Install friction log: C#, Linux

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#4797]

**Comment by hsaliak** [comment]
# gRPC C# on Linux

## Prerequisites
Install mono (can be tricky as the version available as debian package is too old)
Install MonoDevelop (can be tricky as the version available as debian package is too old).

### Step needed for testing unpublished packages only:
Setup local nuget feed - the testing packages from this feed will be used instead of fetching from www.nuget.org 
and copy testing nuget packages (*.nupkg)  into ~/LocalNuGetFeed

```
$ nuget sources  # lists existing sources
$ nuget sources add -Name LocalNuGetFeed -Source $HOME/LocalNuGetFeed

```
Up-to-date testing packages for C# can be downloaded here:
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_packages/platform=windows/82/artifact/artifacts/csharp_nugets.zip

Scenario

- Run MonoDevelop
- Create a new solution  (New solution -> .NET -> .NET console project
- Project -> Add -> Nuget packages
- Install gRPC C# package (choose "local" source if you've setup the local nuget feed)
- Write some code using gRPC. See example below.
- Run the project to make sure everything works.



Example code
(paste this into Program.cs in your new project)
```

using System;
using Grpc.Core;

namespace TestGrpcPackage
{
    class MainClass
    {
        public static void Main(string[] args)
        {
            // This code doesn't do much but makes sure the native extension is loaded
            // which is what we are testing here.
            Channel c = new Channel("127.0.0.1:1000", ChannelCredentials.Insecure);
            c.ShutdownAsync().Wait();
            Console.WriteLine("Success!");
        }
    }
}



```

**Comment by vsubramani** [comment]
Provided instructions don't work.
Code works with 2 workarounds. Details of errors and the workaround provided below.

Environment: Ubuntu with monodevelop

Error 1: 
======
------- Installing...Grpc 0.13.0 -------
Attempting to resolve dependency 'Grpc.Core (≥ 0.13.0)'.
Attempting to resolve dependency 'Ix-Async (≥ 1.2.5)'.
Unable to resolve dependency 'Ix-Async (≥ 1.2.5)'.
System.InvalidOperationException: Unable to resolve dependency 'Ix-Async (≥ 1.2.5)'.

Workaround:
I installed the Ix-Async separately before installing GRPC.

Error 2:
======
System.IO.FileNotFoundException: Error loading native library. File does not exist.
  at Grpc.Core.Internal.UnmanagedLibrary..ctor (System.String libraryPath) [0x00000] in <filename unknown>:0
....
Essentially the file it is looking for is: ...bin/Debug/nativelibs/linux_x64/libgrpc_csharp_ext.so 

Workaround:
I copied over the nativelibs folder from the packages to my bin/debug and it worked.

**Comment by jtattermusch** [comment]
Monodevelop 4.0 is way too old.  How did you manage to restore the nuget packages (AFAIK, there's no nuget support in such old monodevelop).

Did you build using Monodevelop or using xbuild command?  (If you have built from monodevelop, having too old mdtool will likely prevent the native libraries from being copied). Using xbuild might work, but if your mono version is coming from goobuntu, that's also likely too old (it might work, but no guarantees).

**Comment by jtattermusch** [comment]
I can confirm I've been able to go through the instructions on a virtualbox VM (based on ubuntu 14.04), using mono 4.2 and monodevelop 5.10. Everything worked as expected.

**Comment by F4FXL** [comment]
Hi,

I can confirm the above is working.
- Mono 4.5
- Monodevlop 5.10
- Debian Stretch/testing (No VM, real hardware ! :) )

We had to homebrew protoc0.12 because debian package were available for this version and 0.11 has not proto3 support https://github.com/grpc/homebrew-grpc#quick-install-linux

**Comment by jtattermusch** [comment]
@F4FXL, thanks for testing this.
